[MIG] partner_manual_rank: Migration to V15

frahikLV opened the pull request:
Migrate `partner_manual_rank` module and Merge with `partner_customer_supplier`

Reviewers: 
@MiquelRForgeFlow 
@luisg123v
@moylop260

MiquelRForgeFlow committed:
[ADD] partner_manual_rank

oca-travis committed:
[UPD] Update partner_manual_rank.pot

OCA-git-bot committed:
[UPD] README.rst

OpenCode committed:
[MIG] partner_manual_rank: Migration to 14.0

OCA-git-bot committed:
[UPD] README.rst

oca-travis committed:
[UPD] Update partner_manual_rank.pot

francesco-ooops committed:
Added translation using Weblate (Italian)

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (6 of 6 strings)

Translation: partner-contact-14.0/partner-contact-14.0-partner_manual_rank
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-14-0/partner-contact-14-0-partner_manual_rank/it/

pedrocs-exo committed:
Added translation using Weblate (Portuguese)

pedrocs-exo committed:
Translated using Weblate (Portuguese)

Currently translated at 100.0% (6 of 6 strings)

Translation: partner-contact-14.0/partner-contact-14.0-partner_manual_rank
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-14-0/partner-contact-14-0-partner_manual_rank/pt/

jesusVMayor committed:
Added translation using Weblate (Spanish)

jesusVMayor committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (6 of 6 strings)

Translation: partner-contact-14.0/partner-contact-14.0-partner_manual_rank
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-14-0/partner-contact-14-0-partner_manual_rank/es/

frahikLV committed:
[REM] partner_customer_supplier: Remove module

Deprecated infavor of `partner_manual_rank`

moylop260 commented:
@MiquelRForgeFlow @JordiBForgeFlow

Could you review it, please?

celm1990 commented on partner_manual_rank/models/res_partner.py:
```diff
@@ -0,0 +1,64 @@
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    is_customer = fields.Boolean(
+        compute="_compute_is_customer",
+        inverse="_inverse_is_customer",
+        search="_search_is_customer",
+        string="Is a Customer",
+    )
+    is_supplier = fields.Boolean(
+        compute="_compute_is_supplier",
+        inverse="_inverse_is_supplier",
+        search="_search_is_supplier",
+        string="Is a Supplier",
+    )
```
I prefered fields `store=True` as previous version https://github.com/OCA/partner-contact/blob/14.0/partner_manual_rank/models/res_partner.py#L13

frahikLV replied:
This field does not need to be stored, since the information it contains is identical to the information in the *`_rank` field, so it is redundant information. As it has the search method, it behaves identically to a stored field.

What did you think @luisg123v

luisg123v replied:
I think those fields should not be stored for two reasons:

- Since ranks are updated [via SQL](https://github.com/odoo/odoo/blob/120895356a5517ae47146a41539842f0d466256f/addons/account/models/partner.py#L560), such changes won't be identified by dependencies, so computed fields won't be updated.
- Even if they were updated, that would increase chances of concurrency issues. The reason Odoo moved such increasing to SQL is to avoid concurrency issues. But if the "customer" and "supplier" fields are updated as standard fields (by the ORM), the avoided concurrency will come back with those fields

Besides, as @frahikLV mentioned, almost all features are preserved as if the field were stored, as it's searchable.

Regards,

celm1990 replied:
thanks for your comments

luisg123v commented on partner_manual_rank/readme/CREDITS.rst:
```diff
@@ -0,0 +1,2 @@
+ForgeFlow, S.L.
+Vauxoo
```
To the RST syntax, a new line is equivalent to a whitespace. If you want to add several items, you should use a list.

You may check how the resulting README was generated

luisg123v reviewed (changes requested):
I see a few issues regarding commits wording:
- On the removal commit, missing an explanation. Removing a module is not a trivial nor self-explanatory change, so you should mention why, i.e. deprecated infavor of `partner_manual_rank`
- Instead of "module merging", you should say "remove module"
- Since the migration includes some refactor, you should mention on the migration commit what changes are performed, i.e. fields are not stored anymore to avoid computed fields getting outdated and concurrency issues

moylop260 commented:
Hi @MiquelRForgeFlow @JordiBForgeFlow

Friendly reminder to review it, please

frahikLV committed:
[MIG] partner_manual_rank: Migration to V15

Fields are not stored anymore to avoid computed fields getting outdated and concurrency issues.

frahikLV commented:
@luisg123v Your suggestions were applied, could you look again?

moylop260 reviewed (approved):
👍

moylop260 commented:
/ocabot merge nobump

moylop260 commented:
@JordiBForgeFlow

If you have a concern about this PR let me know, please

For now, we will merge it since that it was approved 2 days ago
